plugins/autocomplete/plugin.js

```diff
@@ -0,0 +1,1204 @@
+			this._listeners.push( win.on( 'scroll', function() {
+				this.onChange();
+			}, this ) );
+			this._listeners.push( editorScrollableElement.on( 'scroll', function() {
+				this.onChange();
+			}, this ) );
+
+			this._listeners.push( editor.on( 'contentDom', onContentDom, this ) );
+			// CKEditor's event system has a limitation that one function (in this case this.check)
+			// cannot be used as listener for the same event more than once. Hence, wrapper function.
+			this._listeners.push( editor.on( 'change', function() {
+				this.onChange();
+			}, this ) );
+
+			// Attach if editor is already initialized.
+			if ( editable ) {
+				onContentDom.call( this );
+			}
+
+			function onContentDom() {
+				// Priority 5 to get before the enterkey.
+				// Note: CKEditor's event system has a limitation that one function (in this case this.onKeyDown)
+				// cannot be used as listener for the same event more than once. Hence, wrapper function.
+				this._listeners.push( editable.on( 'keydown', function( evt ) {
+					this.onKeyDown( evt );
+				}, this, null, 5 ) );
+			}
+		},
+
+		/**
+		 * Closes the view and sets its {@link CKEDITOR.plugins.autocomplete.model#isActive state} to inactive.
+		 */
+		close: function() {
+			this.model.setActive( false );
+			this.view.close();
+		},
+
+		/**
+		 * Commits the currently chosen or given item. HTML is generated for this item using the
+		 * {@link #getHtmlToInsert} method and then it is inserted into the editor. The item is inserted
+		 * into the {@link CKEDITOR.plugins.autocomplete.model#range query's range}, so the query text is
+		 * replaced by the inserted HTML.
+		 *
+		 * @param {Number/String} [itemId] If given, then the specified item will be inserted into the editor
+		 * instead of the currently chosen one.
+		 */
+		commit: function( itemId ) {
+			if ( !this.model.isActive ) {
+				return;
+			}
+
+			this.close();
+
+			if ( itemId == null ) {
+				itemId = this.model.selectedItemId;
+
+				// If non item is selected abort commit.
+				if ( itemId == null ) {
+					return;
+				}
+			}
+
+			var item = this.model.getItemById( itemId ),
+				editor = this.editor;
+
+			editor.fire( 'saveSnapshot' );
+			editor.getSelection().selectRanges( [ this.model.range ] );
+			editor.insertHtml( this.getHtmlToInsert( item ), 'text' );
+			editor.fire( 'saveSnapshot' );
+		},
+
+		/**
+		 * Destroys the autocomplete instance.
+		 * View element and event listeners will be removed from the DOM.
+		 */
+		destroy: function() {
+			CKEDITOR.tools.array.forEach( this._listeners, function( obj ) {
+				obj.removeListener();
+			} );
+
+			this._listeners = [];
+
+			this.view.element.remove();
+		},
+
+		/**
+		 * Returns HTML that should be inserted into the editor when the item is committed.
+		 *
+		 * See also the {@link #commit} method.
+		 *
+		 * @param {CKEDITOR.plugins.autocomplete.model.item} item
+		 * @returns {String} The HTML to insert.
+		 */
+		getHtmlToInsert: function( item ) {
```

Comment:
This method force to use `item.name` to be inserted in editor. Where `item.name` is not required parameter. As result of this situation, you cannot use items without name, because those won't be inserted in editor.

Reply:
@msamsel I don't understand what you mean exactly.

1. `item` param is *required* for this mentod.
1. It does not force you to use `item.name` as you're free to override it.

Reply:
@mlewand 
I'm aware that I need to overwrite entire method.
First I thought that there might exists template for inserted content similar to templet for displaying autocomplete item. I hoped that those two parts will work in a similar way.